### PR TITLE
[DOCU-2035] Badge tooltips

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary
Adding tooltips to `FREE`, `ENTERPRISE`, `PLUS`, `OSS ONLY`, and `DB-LESS COMPATIBLE` badges. 

Didn't add a tooltip for the `BETA` badge, as I couldn't figure out what to say that wasn't obvious from the badge label.

### Reason
Lots of confusion around the meaning of the badge labels, particularly the FREE and PLUS badges. 

### Testing
Test in various scenarios:

* In-text labels, for example (need to adjust top margin): [Gateway intro](https://deploy-preview-3523--kongdocs.netlify.app/gateway/)

* H1 headers (page titles): [plugin page](https://deploy-preview-3523--kongdocs.netlify.app/hub/kong-inc/application-registration/) and [main doc page](https://deploy-preview-3523--kongdocs.netlify.app/gateway/2.7.x/vitals/vitals-reports/)

* Tables in main content body, for example in the [contributing guidelines](https://deploy-preview-3523--kongdocs.netlify.app/contributing/markdown-rules/#badges) - last item in table should fit
  
* Tables in plugin sidebar. For example, see the sidebar on [this plugin](https://deploy-preview-3523--kongdocs.netlify.app/hub/kong-inc/application-registration/) (scroll on the right nav)

### CODE REVIEW

File: app/_assets/stylesheets/badges.less [hunk 1]
```diff
@@ -58,25 +54,132 @@
...
+          left: unset;
+          margin-left: 2px;
+          margin-bottom: -5px;
+          top: unset;
+          }
+        }
+      }
+    }
+  //
+  // &:last-of-type {
```
Comment (lena-larionova) [inline app/_assets/stylesheets/badges.less:144]: For whatever reason, this method doesn't work with the current approach (it worked in attempt #1). However, I was able to get around needing it by moving the tooltip above.

**Comment by falondarville** [comment]
I would decrease the space between the badge and the tooltip so that it's more clear what the user is hovering over to get it. As for the overflow issue on the sidebar, I would keep the badge there as it's a recognizable symbol throughout the doc set. I'd have to dig more into adjusting the tooltip to lean left (instead of right) so it doesn't get covered.

**Comment by lena-larionova** [comment]
@falondarville can you take a look again? I think we should consider also moving badges to the line below headers, but for now, tooltips are working!

Please also review the text itself for grammar + let me know if the definitions make sense or need some help. And actually, I might just post this in the Gateway channel, since that's where the concerns came up.

File: app/_assets/javascripts/app.js [hunk 2]
```diff
@@ -683,3 +683,18 @@ $(".closebanner").on("click", function () {
...
+
+
+// Tooltips for badges
+jQuery(function () {
+    $('.badge.enterprise')
+      .append( '<div class="tooltip"><span class="tooltiptext">Available with Enterprise subscription</span></div>' );
+    $('.badge.plus')
+      .append( '<div class="tooltip"><span class="tooltiptext">Available with Plus subscription (Konnect Cloud)</span></div>' );
+    $('.badge.free')
+      .append( '<div class="tooltip"><span class="tooltiptext">Available in Free mode (not open-source)</span></div>' );
```
Comment (flrgh) [inline app/_assets/javascripts/app.js:695]: This is most likely not actionable feedback in the scope of this PR, but I wanted to share it nonetheless:

It's good that we're clarifying `free != oss` now--I think this fills the "I am running OSS, so why can't I turn on feature xyz? The website says it's free" gap. However, I think the average person will then be left wondering

> well, okay, what does "Free mode" mean?

Anecdotally, I was a long-time Kong user before joining the company a couple months ago, and I honestly had no idea what this entailed until I asked about it internally. Googling terms like "kong enterprise free" and clicking around eventually lead me to [this explanatory blog post](https://konghq.com/blog/kong-gateway-enterprise-2-3-now-generally-available/), but I scrolled right past it several times thinking "nah that can't be the information I'm looking for--that's just a blog random blog post." So IMO there's still a communication gap left to bridge here.

File: app/_assets/javascripts/app.js [hunk 3]
```diff
@@ -683,3 +683,18 @@ $(".closebanner").on("click", function () {
...
+    $('.badge.oss')
+      .append( '<div class="tooltip"><span class="tooltiptext" >Available in open-source only</span></div>' );
```
Comment (flrgh) [inline app/_assets/javascripts/app.js:697]: This one IMO could be changed a tad. To me it seems slightly ambiguous and could be mistakenly parsed as:

> This feature is _only_ available in open-source

I think it's better without it "only":

> Available in open-source

Or also some variation of the following:

> Available in Kong OSS
> Available in Kong open-source

File: app/_assets/javascripts/app.js (hunk 2 above)
Comment (falondarville) [reply app/_assets/javascripts/app.js:695]: I would change it to "Available in Enterprise Free mode (without a license)"

File: app/_assets/javascripts/app.js (hunk 3 above)
Comment (lena-larionova) [reply app/_assets/javascripts/app.js:697]: Actually, `OSS ONLY` means exactly that - this is for features that are _only_ available in open-source. This is a very rarely used badge (right now, I think it's only on [this topic](https://deploy-preview-3523--kongdocs.netlify.app/gateway/2.7.x/plan-and-deploy/performance-testing-framework/)).

File: app/_assets/javascripts/app.js (hunk 2 above)
Comment (lena-larionova) [reply app/_assets/javascripts/app.js:695]: ^ I like this one. Will go with that.

**Review by falondarville — APPROVED** [review]
LGTM!

File: app/_assets/javascripts/app.js (hunk 3 above)
Comment (flrgh) [reply app/_assets/javascripts/app.js:697]: ^ relevant to the above but applicable pretty much anywhere: I know you probably feel like you type the work `Kong` 9 million times per day, but I think `Kong OSS` or `Kong open-source` is more readable than `open-source`. Putting `Kong` in front makes it much faster for my brain to parse out "this is a specific noun/thing/product they're referring to" (just like how you've capitalized the `Free` in `Free mode` above, making it clear that `Free` refers to a specific concept and not the generic meaning of `free`).

Comment (lena-larionova) [reply app/_assets/javascripts/app.js]: Still added "Kong" to the name to hopefully clear up _some_ confusion. Hope it helps a little.

**Comment by lena-larionova** [comment]
Re, this point: 

>  Googling terms like "kong enterprise free" and clicking around eventually lead me to this explanatory blog post, but I scrolled right past it several times thinking "nah that can't be the information I'm looking for--that's just a blog random blog post." So IMO there's still a communication gap left to bridge here.

It's a good one. Going to take an action item to add this info properly in the docs somewhere.

**Pull request merged**